### PR TITLE
Add fortls

### PR DESCRIPTION
**Pull request by RMeli**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

**Comment by tldahlgren** [comment]
Adding @adamjstewart since this is a python package.

File: var/spack/repos/builtin/packages/py-fortls/package.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    homepage = "https://fortls.fortran-lang.org"
+    pypi = "fortls/fortls-2.13.0.tar.gz"
+
+    maintainers("RMeli")
+
+    version("2.13.0", sha256="23c5013e8dd8e1d65bf07be610d0827bc48aa7331a7a7ce13612d4c646d0db31")
+
+    depends_on("python@3.7:")
+
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fortls/package.py]: Missing a bunch of deps:
```toml
requires = [
    "setuptools >= 45",
    "wheel",
    "setuptools_scm[toml] >= 6.2",
    "setuptools_scm_git_archive",
]
```

File: var/spack/repos/builtin/packages/py-fortls/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on("py-packaging", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fortls/package.py]: Should be type=build/run

File: var/spack/repos/builtin/packages/py-fortls/package.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    """A Language Server for Fortran providing code completion, diagnostics, hovering and more."""
+
+    homepage = "https://fortls.fortran-lang.org"
+    pypi = "fortls/fortls-2.13.0.tar.gz"
+
+    maintainers("RMeli")
+
+    version("2.13.0", sha256="23c5013e8dd8e1d65bf07be610d0827bc48aa7331a7a7ce13612d4c646d0db31")
+
+    depends_on("python@3.7:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fortls/package.py]: Should probably be type=build/run. However, Spack only supports Python 3.7+, so you could also just remove this line.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-fortls/package.py (hunk 3 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-fortls/package.py]:
```suggestion
```

**Comment by RMeli** [comment]
Thanks @adamjstewart @tldahlgren

File: var/spack/repos/builtin/packages/py-fortls/package.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    maintainers("RMeli")
+
+    version("2.13.0", sha256="23c5013e8dd8e1d65bf07be610d0827bc48aa7331a7a7ce13612d4c646d0db31")
+
+    depends_on("py-setuptools@45:", type=("build", "run"))
+    depends_on("py-wheel", type=("build", "run"))
+    depends_on("py-packaging", type=("build", "run"))
+    depends_on("py-setuptools-scm@6.2:+toml", type=("build", "run"))
+    depends_on("py-setuptools-scm-git-archive", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fortls/package.py]: These should all be type=build except for packaging which is type=build/run

File: var/spack/repos/builtin/packages/py-fortls/package.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    homepage = "https://fortls.fortran-lang.org"
+    pypi = "fortls/fortls-2.13.0.tar.gz"
+
+    maintainers("RMeli")
+
+    version("2.13.0", sha256="23c5013e8dd8e1d65bf07be610d0827bc48aa7331a7a7ce13612d4c646d0db31")
+
+    depends_on("py-setuptools@45:", type=("build", "run"))
+    depends_on("py-wheel", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fortls/package.py]: wheel is also added by the base class so technically you don't need to repeat it

File: var/spack/repos/builtin/packages/py-fortls/package.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on("py-packaging", type=("build", "run"))
+    depends_on("py-setuptools-scm@6.2:+toml", type=("build", "run"))
+    depends_on("py-setuptools-scm-git-archive", type=("build", "run"))
+
+    depends_on("py-json5", type=("build", "run"))
+    depends_on("py-importlib-metadata", type=("build", "run"), when="python@3.7")
+    depends_on("py-typing-extensions", type=("build", "run"), when="python@3.7")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fortls/package.py]:
```suggestion
    depends_on("py-importlib-metadata", type=("build", "run"), when="^python@:3.7")
    depends_on("py-typing-extensions", type=("build", "run"), when="^python@:3.7")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-fortls/package.py (hunk 4 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-fortls/package.py]: Sorry, I thought in the comment above also `setuptools` needed to be of type `run` for some reason. I got confused...

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**